corehq/apps/app_manager/models.py

```diff
@@ -3507,6 +3498,31 @@ def get_filter_value(self, user, ui_filter):
         return None
 
 
+class AncestorLocationTypeFilter(ReportAppFilter):
+    ancestor_location_type_name = StringProperty()
+
+    def get_filter_value(self, user, ui_filter):
```

Comment:
I confirm.  `location.parent.location_type` must be `location.location_type.parent_type`, so you can't have cycles or more than one location in a chain of the same type.